### DISCUSSION TITLE
What should I do if C++11 features are not supported?

### DISCUSSION THREAD

**Discussion by Do-cc** [casadi/casadi#3869] [Q&A]
For some reasons, the supported C++ version is still relatively old and may not be able to use related features above C++11.
I would like to know whether your development relies heavily on these features, and whether you can reduce the compiler requirements by modifying the source code. If so, I would like to know how much impact this will have on the use, including efficiency and stability. Or whether there is a more general version to help me solve related problems. 
Looking forward to your reply.

**Comment by jaeandersson** [comment]
> For some reasons, the supported C++ version is still relatively old and may not be able to use related features above C++11.
> I would like to know whether your development relies heavily on these features, and whether you can reduce the compiler requirements by modifying the source code. If so, I would like to know how much impact this will have on the use, including efficiency and stability. Or whether there is a more general version to help me solve related problems. 
> Looking forward to your reply.

You can always use a newer version of C++ as a CasADi user. C++11 is the oldest supported.
